Packet++/header/VrrpLayer.h

```diff
@@ -0,0 +1,560 @@
+		uint16_t maxAdvInt;
+	};
+
+	/**
+	 * @struct vrrp_packet
+	 * VRRP packet
+	 */
+	struct vrrp_packet {
+#if (BYTE_ORDER == LITTLE_ENDIAN)
+		/** Type */
+		uint8_t type: 4,
+
+		/** Version bits */
+		version: 4;
+#else
+		/** Version bits */
+		uint8_t verson:4,
+
+		/** Type */
+		type: 4;
+#endif
+		/** The Virtual Router Identifier (VRID) field identifies the virtual router this packet is reporting status for*/
+		uint8_t vrId;
+
+		/** This specifies the sending VRRP router's priority for the virtual router */
+		uint8_t priority;
+
+		/** Specifies how many IPvX addresses are present in this Packet */
+		uint8_t ipAddrCount;
+
+		/** This specifies authentication type(v2) or (Max) Advertisement interval (in seconds(v2) or centi-seconds(v3)). */
+		uint16_t authTypeAdvInt;
+
+		/** This specifies checksum field that is used to detect data corruption in the VRRP message.
+		 * VRRPv2 uses normal checksum algorithm, while VRRPv3 uses "pseudo-header" checksum algorithm. */
+		uint16_t checksum;
+
+		/** This specifies one or more IPvX addresses that are associated with the virtual router. */
+		uint8_t *ipAddresses[];
+	};
+
+	/**
+	 * VRRP versions
+	 */
+	enum VrrpVersion {
+		/** Unknown VRRP version */
+		/** VRRP version 2 */
+		Vrrp_Version_2 = 2,
+
+		/** VRRP version 3 */
+		Vrrp_Version_3 = 3
+	};
+
+	/**
+	 * VRRP message types
+	 */
+	enum VrrpType {
+		/** VRRP advertisement packet */
+		VrrpType_Advertisement = 1
+	};
+
+	/**
+	 * @class VrrpLayer
+	 * A base class for all VRRP (Virtual Router Redundancy Protocol) protocol classes. This is an abstract class and cannot be instantiated,
+	 * only its child classes can be instantiated. The inherited classes represent the different versions of the protocol:
+	 * VRRPv2 and VRRPv3
+	 */
+	class VrrpLayer : public Layer {
+	private:
+		/**
+		* A method that add IP Addresses to data offset
+		* @param ipAddresses [in] IP Address vector
+		* @param offset [in] Data offset
+		* @return ture if add successfully, false otherwise.
+		*/
+		bool addIPAddressesAt(const std::vector<IPAddress> &ipAddresses, int offset);
+
+		IPAddress::AddressType m_addressType;
+
+	public:
+		VrrpLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet, ProtocolType vrrpVer,
+				  IPAddress::AddressType addressType)
+				: Layer(data, dataLen, prevLayer, packet) {
+			m_Protocol = vrrpVer;
+			m_addressType = addressType;
+		}
+
+		/**
+		 * A constructor that allocates a new VRRP
+		 * @param[in] subProtocol The VRRP protocol
+		 */
+		explicit VrrpLayer(ProtocolType subProtocol);
+
+		virtual ~VrrpLayer() {}
+
+		/**
+		* A method that get IP Address type
+		* @return The VRRP IP Address type
+		*/
+		IPAddress::AddressType getAddressType() const;
+
+		/**
+		* A method that set IP Address type
+		* @param addressType [in] IP Address type
+		*/
+		void setAddressType(IPAddress::AddressType addressType);
```

Comment:
What's the point in exposing a public method to set the address type? 🤔

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1118/commits/9fbadc5322818d2c02831a7aac01b3577fa14267